Yank/repex.py

```diff
@@ -159,21 +251,57 @@ def open(self, mode='r'):
                 ncfile.Conventions = 'ReplicaExchange'
                 ncfile.ConventionVersion = '0.2'
                 ncfile.DataUsedFor = nc_name
-                ncfile.checkpoint_interval = self._checkpoint_interval
+                ncfile.CheckpointInterval = checkpoint_interval
+
+                # Create and initilize the global variables
+                nc_last_good_iter = ncfile.createVariable('last_iteration', int, 'scalar')
+                nc_last_good_iter[0] = 0
+                return True
+            else:
+                return False
+
+        # Setup the primary file if needed
+        # Generate a unique UUID in case we need to assign it
+        # primary and subfiles are linked by UUID
+        # Uses uuid4 to avoid assigning hostname information
+        primary_uuid = str(uuid.uuid4())
+        ncfile_ever_built = False
+        for nc_name, ncfile in utils.dictiter(primary_ncfiles):
+            ncfile_built = check_and_build_storage_file(ncfile, nc_name, self._checkpoint_interval)
+            if ncfile_built:
+                ncfile_ever_built = True
+            # Assign ncfile to class property
+            setattr(self, '_storage_' + nc_name, ncfile)
+        if ncfile_ever_built:  # Assign the same UUID to all primary files
+            for _, ncfile in utils.dictiter(primary_ncfiles):
+                ncfile.UUID = primary_uuid
+        else:
+            primary_uuid = primary_ncfiles['analysis'].UUID
+        # Handle Subfiles
+        for nc_name, ncfile in utils.dictiter(sub_ncfiles):
+            # If they are in the sub_ncfiles dict, they exist and are open
+            ncfile_built = check_and_build_storage_file(ncfile, nc_name, self._checkpoint_interval)
+            if ncfile_built:
+                # Assign the UUID to the subfile
+                ncfile.UUID = primary_uuid
             setattr(self, '_storage_' + nc_name, ncfile)
-        on_file_interval = self._storage_checkpoint.checkpoint_interval
-        if on_file_interval != self._checkpoint_interval:
-            logger.debug("checkpoint_interval != on-file checkpoint interval! "
-                         "Using on file checkpoint interval of {}.".format(on_file_interval))
-            self._checkpoint_interval = on_file_interval
+
+        if self._storage_analysis is not None:
```

Comment:
No, it could earlier when I had it check the checkpoint file, but I realized that we may need the checkpoint information for non-checkpoint file operations and changed the target file, I'll remove it